src/components/Nav/Navigation.tsx

```diff
@@ -106,25 +48,42 @@ class Navigation extends React.Component<PropsType, StateType> {
   }
 
   setControlledState = event => {
-    if (event.activePath) {
-      // keep track of path as user clicks on nav bar
-      this.setState({ selectedItem: event.activePath });
-    } else if ('navCollapsed' in event) {
+    if ('navCollapsed' in event) {
       this.props.setNavCollapsed(this.props.navCollapsed);
     }
   };
 
+  renderMenuItems() {
+    const { location } = this.props;
+    const menuItems = routes();
+    const activeItem = menuItems.find(item => {
+      let isRoute = _.startsWith(location.pathname, item.to);
+      if (!isRoute && item.pathsActive) {
+        isRoute = _.filter(item.pathsActive, path => path.test(location.pathname)).length > 0;
```

Comment:
I think you can use [matchPath](https://reacttraining.com/react-router/web/api/matchPath), this might be safer than our current `startsWith` approach.
Also, maybe this example can be useful: https://reacttraining.com/react-router/web/example/custom-link

Reply:
Changed